app/main.py

```diff
@@ -1,4 +1,48 @@
-KNIGHTS = {
+from __future__ import annotations
+
+from app.knights.knight import Knight
+
+
+# creation of knights
+
+
+def knights_config(knights: dict[str, any]) -> dict:
+    return {knight_name: Knight(
+            knight_data["name"],
+            knight_data["power"],
+            knight_data["hp"],
+            knight_data["armour"],
+            knight_data["weapon"],
+            knight_data["potion"]) for
+            knight_name, knight_data in knights.items()}
+
+
+# BATTLE
+def battle(knights: dict[str, any]) -> dict[str, int]:
+    ready_knights = knights_config(knights)
+    for attacker, defender in [
+        ("lancelot", "mordred"),
+        ("arthur", "red_knight")
+    ]:
+        attacker_knight = ready_knights[attacker]
+        defender_knight = ready_knights[defender]
+
+        defender_knight.hp -= (
+            attacker_knight.power - defender_knight.protection
+        )
+        attacker_knight.hp -= (
+            defender_knight.power - attacker_knight.protection
+        )
+
+        Knight.check_hp(attacker_knight)
```

Comment:
```suggestion
        attacker_knight.check_hp()
```

Reply:
Why you need to call as class if you can call to instance?
The same with defender

Reply:
I don't know indeed )